connection_go18.go

```diff
@@ -0,0 +1,24 @@
+// +build go1.8
+
+package mysql
+
+import (
+	"context"
+)
+
+func (mc *mysqlConn) Ping(ctx context.Context) error {
+	err := mc.writeCommandPacket(comPing)
+	if err != nil {
+		return err
+	}
+
+	select {
+	case <-ctx.Done():
+		mc.Close()
+		return ctx.Err()
+	default:
+	}
+
+	_, err = mc.readResultOK()
```

Comment:
I'm not sure what you mean. Are you saying we should still read the OK result after the Done channel is closed? If so, what's the point if we're closing the connection anyway? And if not, please explain more.